### ISSUE TITLE
Apple new Enterprise certificates : got "The signing certificate is not from the expected issuer"

### ISSUE THREAD

**Issue by lamselli** [airsdk/Adobe-Runtime-Support#511]
Hello !

When trying to build an IPA with new entreprise distribution certificate, I get the following error message from adt : "The signing certificate is not from the expected issuer"

![screen_capture](https://user-images.githubusercontent.com/5931359/98163300-ecd8b780-1ee2-11eb-8941-4da15c5c849c.jpg)

It seems that I am not the only one with this problem : https://forum.starling-framework.org/d/22729-the-signing-certificate-is-not-from-the-expected-issuer

It seems to be due to the fact that the new certificates use a new Apple intermediate certificate.

I am using AIR 33.1.1.190 under macOS Mojave (10.14.6).

Using CodeSign, I managed to resign some old IPA but I can't package any new ones...

Has anyone who experienced the same issue found a workaround ?

Is it possible to make a quick fix with adt regarding this problem ?

Thank you !

**Comment by ajwfrost** [comment]
Hi
We've been looking at this, and could do with confirming a few details - are you able to contact me at Andrew.Frost at Harman.com?
thanks

**Comment by FlorenceLoukas** [comment]
Hello,
I'm having exactly the same problem with new apple certificates

![Capture](https://user-images.githubusercontent.com/43169837/98951439-89f6aa00-24fa-11eb-8b19-4cf9e30d8cec.JPG)


Can you tell me how did you solve it ?

**Comment by lamselli** [comment]
Bonjour Florence,

Pour le moment, je n'ai pas résolu mon problème.
Je suis sur Mac, donc j'ai pu suivre les conseils du [ticket 504](https://github.com/Gamua/Adobe-Runtime-Support/issues/504), mais pour autant je n'ai pas réussi à tout faire fonctionner...
J'attends des nouvelles d'Andrew par email pour essayer de debuger le souci.

@ajwfrost : It seems that Florence is french like me. So I answered in french. I'm saying that for the time being I'm still experiencing the issue and that you should send me a debug adt shortly to try to find a fix. Is it correct ?
Thanks !

**Comment by ajwfrost** [comment]
Thanks @lamselli  - although, we did already send you an ADT file with debug output... let me check in case that got blocked by email servers...

**Comment by FlorenceLoukas** [comment]
Thank you both !
@ajwfrost I just recieve your mail with your instructions, i'm going to try and let you know.

**Comment by reckhoff** [comment]
@ajwfrost - will this ADT file help me on #504? happy to give it a try if you want to send.

**Comment by ajwfrost** [comment]
It's still work in progress currently .. the test version of ADT has the right issuer certificate details in it now, but when the IPA files are signed, they're ending up not working properly...

**Comment by lamselli** [comment]
Hey @ajwfrost !
Did you manage to make any progress regarding this issue ?

Thanks for your feedback !

**Comment by FlorenceLoukas** [comment]
Hello @ajwfrost, 
Any news for us ?
Let us know

**Comment by lamselli** [comment]
Hey @ajwfrost,
Still stuck with this issue...
Can you give us any news ?

**Comment by ajwfrost** [comment]
Hi
Currently we're not able to reproduce this, we're trying to look into it but having checked some of the errors, it actually looks like the certificate used for signing isn't then working on the handset: not sure if it's then the handset that doesn't have the appropriate trust enabled for the signing certificate's issuer..?
Are you able to use these new certificates for a normal iPhone application? or can you re-sign one of the new IPA files with this new certificate and have it work?
thanks
  Andrew

**Comment by lamselli** [comment]
I am not sure that I understand your question.

Currently, if I build with a developer certificate and a developper mobile provisioning profile, I can install on my device.

With the enterprise distribution certificate and a distribution provisioning profile, if I just build with the debug adt you gave me, I get no error, but I can't install the application on any IOS device.
As I told you before, I need to resign the ipa (I'm not changing the provisioning profile) using the following commands : 
#unzip the ipa :
unzip Main.ipa
#export the entitlements from the unzipped pakage:
codesign -d --entitlements :entitlements.plist Payload/AppName.app/
#delete previous signature folder :
rm -rf Payload/AppName.app/_CodeSignature
#resign using the exact same Distribution certificate (but with codeSign instead of using exported p12 file from this certificate)
codesign --verbose -d -f -s "iPhone Distribution: **********" --entitlements entitlements.plist Payload/Familink.app
#rezip the resigned ipa
zip -gr resigned.ipa Payload

After this manipulation, I can install the application on any IOS device (that approved my entreprise account of course)

Did you also receive my email on 12 nov. 2020 18:04 with both _CodeSignature folders ? The one that is working, and the other that isn't ? Did it help?

**Comment by ajwfrost** [comment]
Ah - okay thanks, this is what I was looking for, I think I missed the email from 12th November (or had sent the wrong thing over to our developer perhaps, he said he only had one file..) - I'll dig that out and we can check.

The step-by-step at least shows us that it should be possible to sort this out and it should work then with this certificate... thanks!

**Comment by lamselli** [comment]
Hi Andrew,

For your information, I just sent you an email with the example apps as you requested. Just tell me if you received my email correctly, and good luck with working on the issue !

**Comment by lamselli** [comment]
Hello @ajwfrost,
Any news/progress with this issue ?

**Comment by ajwfrost** [comment]
Hi - we've got a few issues with certificates and code-signing currently being worked on, so this should be solved as part of this. For the short term, are you able to cope with the re-signing process suggested on 9th December, above?
thanks

**Comment by FlorenceLoukas** [comment]
Unfortunately not : I work on a PC

**Comment by mc3dev** [comment]
Hi, I am on PC using Adobe Animate with AIR 33.0.2.246 for iOS. I am having the same "The signing certificate is not from the expected issuer" error. Is there a way to sign the IPA via Adobe Animate? Can you please provide some help? Thank you so much!
![image](https://user-images.githubusercontent.com/5835724/106509798-0f577e00-649c-11eb-8872-5bfb9c47ef02.png)

**Comment by mc3dev** [comment]
> Hi - we've got a few issues with certificates and code-signing currently being worked on, so this should be solved as part of this. For the short term, are you able to cope with the re-signing process suggested on 9th December, above?
> thanks

Is there an ETA for a resolution?

I re-signed the IPA using the Dec 9th instructions post but when the client add the IPA to InTune MDM it threw this error message "The selected app package does not appear to have an app id." Do you have any insight about this error?
![error](https://user-images.githubusercontent.com/5835724/106861370-ac303c00-6693-11eb-896c-c56ad51d8ac5.png)

**Comment by Umanoff** [comment]
Will there continue to be a workflow on Windows, and Windows only, for using iOS Enterprise certificates to sign AIR apps?

**Comment by ajwfrost** [comment]
Hi - I think I missed some notifications from this thread, apologies for the slow response. We're hopeful that we can get this working, on Windows as well as Mac.. it's very complex but I think we can see how this will work. Not in time for our next release but as soon as we have this, we'll push out an update (it would just be the adt.jar file that does all this..).

**Comment by StefanTasklift** [comment]
I'm having the same problem with my iOS development certificate. My previous certificate expired and when I created a new one the problem occurred. 

So I guess it's a general problem related to all certificates created with Apples new intermediate certificate. Not just Enterprise.

Great to se that Harman is taking this seriously and are working on a solution :)

**Comment by lamselli** [comment]
Actually, Apple said that :
Starting January 28, 2021, the digital certificates you use to sign your software for installation on Apple devices, submit apps to the App Store, and connect to certain Apple services will be issued from a new intermediate Apple Worldwide Developer Relations certificate

So I think that now all certificates will experience the issue whereas only entreprise certificate were having it before...

**Comment by Dima107** [comment]
Hi we had same issue with new Apple's certificate: _The signing certficate is not from the expected issuer_
MacOS 10.14.6, AIR SDK 33.1.1.300
Now I use AIR SDK 33.1.1.406. I built IPA file. But I can't install IPA. I have error and warnings

_WARNING: could not locate iTunesMetadata.plist in archive!
WARNING: could not locate Payload/MotoX3M.app/SC_Info/MotoX3M.sinf in archive!
ERROR: Install failed. Got error "ApplicationVerificationFailed" with code 0xe8008001: Failed to verify code signature of /var/installd/Library/Caches/com.apple.mobile.installd.staging/temp.J12mUb/extracted/Payload/MotoX3M.app : 0xe8008001 (An unknown error has occurred.)_

I have distribution certificate only
Have someone same issues?
Who can help me to solve this?

**Comment by ajwfrost** [comment]
Yes there is still a problem with getting code-signing working here, it's causing quite a headache as it's just not documented so we're having to work via a lot of reverse-engineering a binary that has been properly signed..

On MacOS you can re-sign the application (please scroll up to 9th December: https://github.com/Gamua/Adobe-Runtime-Support/issues/511#issuecomment-741844598)

I'm thinking that actually it may just be easier for us to push this code-signing task onto the OS tools anyway, where they exist i.e. on MacOS. The changes we're then doing would just be for developers using Windows to create IPA files..

**Comment by Umanoff** [comment]
@lamselli Could you please confirm that the steps that you described in your Dec 9 message here actually produce a working ipa.
I think that you are saying to use Apple's ___.cer certificate and NOT a passworded .p12 based on the ___.cer file. Is that correct? 
I'm confused because you wrote: "...but with codeSign instead of using exported p12...":  Are we not using codesign in any event? Forgive me, but I'm coming from a Windows environment, trying to understand the Mac workflow.

**Comment by lamselli** [comment]
I cannot guarantee that my workflow will work on the app store since I used it with an entreprise account.
All I can say is that I managed to install the resigned ipa on several iphones.
I built the ipa as if everything was working (using a debug version of adt that Andrew sent me by email)
The ipa won't install on any device, so I use the codesign command line to resign the ipa, using the workflow described in my message.
(The command codesign --verbose -d -f -s "iPhone Distribution: **********" --entitlements entitlements.plist is indeed telling to sign the ipa using the specified certificate (same name that is in your keychains) whereas adt is using the p12 certificate to sign, and it seems that it has an issue...
Hope it is clearer for you (and sorry if my English is not clear, it is not my native language)

**Comment by jimmymjing** [comment]
I'm getting the same issue: WARNING: could not locate Payload/BioFanta.app/SC_Info/BioFanta.sinf in archive!
Will roll back to the previous version AIR 33.1.1.345 solve the issue for the time being?

**Comment by ajwfrost** [comment]
That warning is just a warning, it shouldn't be the thing that's preventing the install.
The underlying issues that we're working on:
a) initially "the signing certificate is not from the expected issuer" because ADT checks that your certificates have the right Issuer field per the certificates that we have internally set as trusted; we've got the new intermediate certificate in ADT now so this error has gone away
b) requirements to use version 20400 for the CodeDirectory format in the file signature -> ongoing and pretty much done
c) correct PKCS7 signature using the new Apple cert chain -> working on this currently, making slow progress

If we don't get it working at least for simple cases by the end of this week then we will look at providing a workaround for the MacOS platforms..

**Comment by jimmymjing** [comment]
@ajwfrost Thanks for making the progress clear. I'm on Windows and couldn't install compiled ipa with either debugging or ad-hoc certificates on devices. I will just wait for the next update then!

**Comment by Mobiescape** [comment]
Hi,

We are also having the same issue and looking for the solution here. Andrew you are really great. I hope you will succeed with this issue. waiting for an update. Thanks.

**Comment by Klug76** [comment]
Hi
With version 406 we cannot install and run ipa (tried from windows and from macosx).
Error:
```
[java] WARNING: could not locate iTunesMetadata.plist in archive!
     [java] WARNING: could not locate Payload/MobileTest.app/SC_Info/MobileTest.sinf in archive!
     [java] ERROR: Install failed. Got error "ApplicationVerificationFailed" with code 0xe8008001: Failed to verify code signature of /var/installd/Library/Caches/com.apple.mobile.installd.staging/temp.tEiqi0/extracted/Payload/MobileTest.app : 0xe8008001 (An unknown error has occurred.)
     [java]
```
Hope for fix...

**Comment by Umanoff** [comment]
@lamselli I cannot quite understand your Codesign workflow because I have been developing AIR apps on Windows. We don’t have this ‘entitlements.plist’ thing in Windows, so I need to ask: in the example that you give, above, you use this line:
codesign --verbose -d -f -s "iPhone Distribution: **********" --entitlements entitlements.plist Payload/Familink.app .
What is the ‘**********’? Is that the name of the signing certificate issued by Apple? With or without the ‘.cer’ file extension?
And what is ‘Familink.app’? Is that the name of the app you want to sign (and that you previously called ‘AppName.app’ ?

Clarity is important in these things!

And, to be clear, you are saying that this solution to re-signing with Codesign is working for you with an Enterprise Account distribution certificate? With an app that was originally built with an ‘In-House’ provisioning profile?

I’m asking all of these questions because, if your solution works, I’ll go out and buy a Macbook Pro today!  

Thanks, Craig

**Comment by Umanoff** [comment]
@ajwfrost  I’m sure that your people are trying their best with this problem but, for those of us who have to manage clients whose AIR-made iOS apps have failed, can you give us your best estimate of the time to enable, once again, iOS development via Harman AIR on either  Windows or on Mac?  Days? Weeks? Months? Ever?

This will let us know what to tell our clients (in my case one impatient Fortune-500 client); and will let us plan the way we will resume app development – either with AIR (my preferred way) or with native systems.

**Comment by ajwfrost** [comment]
Hi

The 'entitlements' thing is built into the binary, those workaround steps were first to pull out the existing entitlements and then to use this to re-sign with the same entitlements. And the '****' parts were to hide the details of his signing certificate (this is a string alias that's used for the code-signing, rather than the actual file). And yes, the extracted application would be 'something'.app, depends on what the application name was. So, very much mac-based...

On the plus side: we have apparently cracked at least the basic signing - not 100% sure whether this will work with ANEs and we know that Swift libraries will be a separate hurdle, but in case it helps, we are building release version of ADT and I'll post that here once it's ready in a few hours. Perhaps we should call this a beta: there's one 'magic number' that we're not happy with, so need to check whether we need that or can find out where it should come from..

So I'd suggest holding off buying a macbook until tomorrow...

thanks

**Comment by lamselli** [comment]
@Umanoff  I'll try to be as clear as I can.
Here are the steps I use to build an IPA with my macbook that is correctly installed with an entreprise provisioning profile (so for in-house deployment)

First I build my ipa using IntelliJ IDEA (but I believe any method would be OK)
I get a Main.ipa file.

Then I open a terminal and I unzip the ipa using the unzip command:
unzip Main.ipa
It will generate a folder named Payload, and under this folder another folder named AppName.app (in my case Familink.app)

I export the entitlements from the unzipped package using codesign (I don't know if this step is usefull, but I followed all the steps described there : https://stackoverflow.com/questions/6896029/re-sign-ipa-iphone) 👍 
codesign -d --entitlements :entitlements.plist Payload/AppName.app/ (in my case : codesign -d --entitlements :entitlements.plist Payload/Familink.app/)
It will generate an entitlements.plist file.

I delete the previous signature folder (Payload/AppName.app/_CodeSignature), in my case:
rm -rf Payload/Familink.app/_CodeSignature

I resign the full application folder (Payload/AppName.app) using the exact same Distribution certificate that I used in Intellij except that in intellij I gave a .p12 file exported from my keychain, and with codeSign I will give the exact same name as given in my keychain (with quotes " ") with the following command in my case :
codesign --verbose -d -f -s "iPhone Distribution: MyClientsCompanyName" --entitlements entitlements.plist Payload/Familink.app

Once the signature process is done, I rezip the resigned Payload folder to make it back to an ipa file : 
zip -gr resigned.ipa Payload

I get a "resigned.ipa" file that I can install on any IOS device.

So if you're stuck and that you need a quick workaround. I cannot garantee that it will work for you, but it worked for me...

Hope this comment will help you

**Comment by ajwfrost** [comment]
FYI in case people want to try this:
https://transfer.harman.com/message/vKmMMoGcl3KMvyv0X0NtDv
I would be interested to hear the results, we've tried this on a few things here and will continue to look at further testing/refinement as necessary..

thanks

**Comment by lamselli** [comment]
I replaced adt.jar file and got the following error (I was using AIR_SDK_33.1.1.300): 
unexpected failure: Mismatch between ADT and runtime classes
java.lang.IllegalStateException: Mismatch between ADT and runtime classes
at com.adobe.air.apk.APKOutputStream.f(APKOutputStream.java)
at com.adobe.air.apk.APKOutputStream.n(APKOutputStream.java)
at com.adobe.air.apk.APKOutputStream.addApplicationDescriptor(APKOutputStream.java)
at com.adobe.air.ApplicationPackager.addSpecialFiles(ApplicationPackager.java)
at com.adobe.air.ApplicationPackager.createPackage(ApplicationPackager.java)
at com.adobe.air.ADT.parseArgsAndGo(ADT.java)
at com.adobe.air.ADT.run(ADT.java)
at com.adobe.air.ADT.main(ADT.java)

**Comment by ajwfrost** [comment]
Yes, sorry this was just for testing the IPA creation, with Android there needs to be a version match between the different files otherwise things go awry...

**Comment by lamselli** [comment]
It seems to be working for me. (I have done too many tests, I just built an ipa and installed it on my iPhone, and it launched.

**Comment by Jasenvm** [comment]
Hi Guys

I thought I was going crazy with this issue for two days now, after everything working on windows 10, Flex Mobile/Lastest IntelliJ Android app...so, I started deploying to my new mac / iPad IOS 14 setup / MAC 11.2.1. Only to stumble upon this thread and feel so relieved this fix is actively been worked on.

Just word of encouragement and confidence in the future of the Harman AIR / SDK platform!

I'll try this solution out too.

Thanks, @ajwfrost, and Team.

**Comment by Elintondm** [comment]
Using this new adt.jar with AIR_SDK_33.1.1.406 I got this error building my ipa:

`java.lang.RuntimeException: ADT error:
unexpected failure: null
java.lang.NullPointerException
	at com.adobe.air.ipa.MachoSigner.newRequirementsBlob(MachoSigner.java)
	at com.adobe.air.ipa.MachoSigner.sign(MachoSigner.java)
	at com.adobe.air.ipa.MachoSigner.sign(MachoSigner.java)
	at com.adobe.air.ipa.IPASigner.signMacho(IPASigner.java)
	at com.adobe.air.ipa.IPAOutputStream.finalizeFrameworkSig(IPAOutputStream.java)
	at com.adobe.air.ApplicationPackager.createPackage(ApplicationPackager.java)
	at com.adobe.air.ipa.IPAPackager.createPackage(IPAPackager.java)
	at com.adobe.air.ADT.parseArgsAndGo(ADT.java)
	at com.adobe.air.ADT.run(ADT.java)
	at com.adobe.air.ADT.main(ADT.java)
	at com.fontec.generator.ClientBuilder._runBuildBat(ClientBuilder.java:1037)
	at com.fontec.generator.ClientBuilder.runBuildBat(ClientBuilder.java:946)
	at com.fontec.generator.ClientBuilder.build(ClientBuilder.java:563)
	at com.fontec.generator.ClientBuilder.checkProcess(ClientBuilder.java:344)
	at com.fontec.generator.ClientBuilder.processBrandsAndPlatforms(ClientBuilder.java:353)
	at com.fontec.generator.ClientBuilder.process(ClientBuilder.java:244)
	at com.fontec.generator.Builder.start(Builder.java:277)
	at com.fontec.generator.Main.lambda$main$0(Main.java:95)
	at java.util.ArrayList.forEach(ArrayList.java:1249)
	at com.fontec.generator.Main.main(Main.java:85)`

Which sdk are you using?

**Comment by ajwfrost** [comment]
@Elintondm thanks, that's not expected .. it doesn't tell us exactly where the null reference exception is happening but it is probably either that it can't find the application identifier (!) or an issue with the signing certificate. I suspect we may need to provide a version with more debugging, to track that down... will try to get that published tomorrow.

thanks

**Comment by pavelzmolik** [comment]
@ajwfrost Thank you very much for the new ADT.jar
I tried to compile with the 33.1.1.406 and new adt.jar and IPA is installed correctly, and codesign is 20400, so there is not the warning alert about upgrading application..
Good job guys!
Hope u deploy new public version od SDK soon.

**Comment by ajwfrost** [comment]
Thanks @pavelzmolik 
@Elintondm  would you be able to try again with the below version, it should have a fix for your issue and also has some debug output to help us see if there are other issues...
https://transfer.harman.com/message/vLzOyqAyGmtFkIjXkKV7Kh
(will also provide a version of this without the debug soon, as I think that anything we put out on stdout may cause other tools like Animate to think we've failed..)

**Comment by Elintondm** [comment]
Thanks, @ajwfrost !

With this new version the ipa was generated but can't be installed. I will get the error log and post it here.

**Comment by ajwfrost** [comment]
@Elintondm  as well as the error log .. would it be possible to get the IPA itself? you can upload it using the below link:
https://transfer.harman.com/requests/4T5CYQs45b5HzU6pJ3cXgO
thanks

**Comment by Elintondm** [comment]
Hi @ajwfrost !

Files Sent!

**Comment by ajwfrost** [comment]
Thanks @Elintondm, appreciate that! - yes so we can look at that FBAudienceNetwork framework to see what's wrong with the code signature...

**Comment by nygas** [comment]
> Thanks @pavelzmolik
> @Elintondm would you be able to try again with the below version, it should have a fix for your issue and also has some debug output to help us see if there are other issues...
> https://transfer.harman.com/message/vLzOyqAyGmtFkIjXkKV7Kh

Hi ajwfrost.
I also had a problem with installing the .ipa file (compilation was fine). After replacing the adt.jar file in version 33.1.1.406, I was finally able to install the application (from Windows 10) on my iPad mini 2.
Thx!

**Comment by ajwfrost** [comment]
@Elintondm are you able to try another version please, this one should fix the framework signing..
https://transfer.harman.com/message/aQcMhpbLe1VqxU8cq4G6lf
thanks

**Comment by Elintondm** [comment]
> @Elintondm are you able to try another version please, this one should fix the framework signing..
> https://transfer.harman.com/message/aQcMhpbLe1VqxU8cq4G6lf
> thanks

Thanks @ajwfrost ! 

It worked!!!

**Comment by ajwfrost** [comment]
@Elintondm great, thanks for the confirmation! So, I guess we should get all this into a proper release of the AIR runtime shortly... (and then I think we have the same sort of challenge to face on Android...)

Thanks

**Comment by pavelzmolik** [comment]
@ajwfrost 
Hello, please check in the next public version if there is not issue with distribution certificates and profiles.. I have a info from my bussines partner.. He downloaded SDK 406 with ADT.jar fix u provide and tried compile app with distribution certificates and profiles.. And Apple store refused that IPA because of signing problems. I told him you probably make debug ADT.jar  to test if there is not signing issue with developer certificates.
So he returned back to v.345 But I hope in next public release this you will test and fix with distribution profiles.. Its just for your informations
thanks a best regards

**Comment by ajwfrost** [comment]
Hi @pavelzmolik 
Can I check a few things please:
1. does it work for him with v.345?
2. are you using a certificate that was generated recently? are you able to check for the details of the 'issuer' for this certificate via the Keychain tool and let us know what that is?
3. would it be feasible for you to send us a couple of packages - firstly the IPA generated with the SDK 406 + updated adt.jar, and secondly an IPA that's created from this first one but re-signed using the workaround steps posted up above on Dec 9th?
4. do you have the details from Apple for exactly what the problem was?

If you need me to provide a link for you to upload files, please let me know..

thanks

**Comment by megajogos** [comment]
Hi @ajwfrost !

With your last adt.jar I was able to build and install the IPA. But when I summit to itunes I got the following message:

`We identified one or more issues with a recent delivery for your app, "MegaJogos Cartas e Tabuleiro" 105.1.3 (105.14.3). Please correct the following issues, then upload again.

ITMS-90428: Invalid Swift Support - The files libswiftFoundation.dylib, libswiftos.dylib, libswiftMetal.dylib, libswiftDispatch.dylib, libswiftUIKit.dylib, libswiftObjectiveC.dylib, libswiftCoreGraphics.dylib, libswiftDarwin.dylib, libswiftCoreFoundation.dylib, libswiftCore.dylib, libswiftCoreImage.dylib, libswiftQuartzCore.dylib don’t match /Payload/MegaJogos.app/Frameworks/libswiftFoundation.dylib, /Payload/MegaJogos.app/Frameworks/libswiftos.dylib, /Payload/MegaJogos.app/Frameworks/libswiftMetal.dylib, /Payload/MegaJogos.app/Frameworks/libswiftDispatch.dylib, /Payload/MegaJogos.app/Frameworks/libswiftUIKit.dylib, /Payload/MegaJogos.app/Frameworks/libswiftObjectiveC.dylib, /Payload/MegaJogos.app/Frameworks/libswiftCoreGraphics.dylib, /Payload/MegaJogos.app/Frameworks/libswiftDarwin.dylib, /Payload/MegaJogos.app/Frameworks/libswiftCoreFoundation.dylib, /Payload/MegaJogos.app/Frameworks/libswiftCore.dylib, /Payload/MegaJogos.app/Frameworks/libswiftCoreImage.dylib, /Payload/MegaJogos.app/Frameworks/libswiftQuartzCore.dylib. Make sure the files are correct, rebuild your app, and resubmit it. Don’t apply post-processing to /Payload/MegaJogos.app/Frameworks/libswiftFoundation.dylib, /Payload/MegaJogos.app/Frameworks/libswiftos.dylib, /Payload/MegaJogos.app/Frameworks/libswiftMetal.dylib, /Payload/MegaJogos.app/Frameworks/libswiftDispatch.dylib, /Payload/MegaJogos.app/Frameworks/libswiftUIKit.dylib, /Payload/MegaJogos.app/Frameworks/libswiftObjectiveC.dylib, /Payload/MegaJogos.app/Frameworks/libswiftCoreGraphics.dylib, /Payload/MegaJogos.app/Frameworks/libswiftDarwin.dylib, /Payload/MegaJogos.app/Frameworks/libswiftCoreFoundation.dylib, /Payload/MegaJogos.app/Frameworks/libswiftCore.dylib, /Payload/MegaJogos.app/Frameworks/libswiftCoreImage.dylib, /Payload/MegaJogos.app/Frameworks/libswiftQuartzCore.dylib.

Best regards,`

**Comment by ajwfrost** [comment]
Thanks @megajogos - @pavelzmolik are you also getting swift-related issues?
This makes sense as I can see the .dylib handling is still lacking - the code-signature blob isn't being correctly set up for the latest format, for these libraries. I think we've worked out what we need to do to fix this (but I just got engrossed and forgot to feed my kids .. will try to get an updated version out tomorrow...)

**Comment by pavelzmolik** [comment]
@ajwfrost 
Hello, i put the questions you gave here to my bussines partner...
He answered me
1/ version 345 is properly working if he upload IPA into appstore(appstore itself resigned the IPA files with distribution certificates.. method of resign  of appstore is undigable for me now)
2/ certificates are quite old.. maybe this is the main problem why the distribution certificates od my partner dont work.. He promised me this week he tries make new certificates and get compile IPA with them and try make upload.. I give u a result here when I get a feedback from him..
3/ this will be affected byt point 2/ .. FYI we always use the current public release for compile distribution IPA as I cant say he is compiled it on Windows or Mac.. but it seems it is doesnt matter if you compile it on Windows or Mac
4/ If there will be some detailed issues form appstore.. I asked to resend that issues to me and I can put them here.. to debug it..

Man I really appreciated work you done here.. I know nobody have same compiller and same settings... I admire your enthusiasm to fix all bugs that incomplete logs are here.. I really think AIR is one of the best SDK for using one asset for multiple platforms you can use if you dont want make couple of png sprite sheets..
I write here when I dig some news..

**Comment by ajwfrost** [comment]
Hi @megajogos @pavelzmolik 
Thanks for the feedback: we have another update to ADT.jar that should fix the signing of the Swift libraries:
https://transfer.harman.com/message/Am40qkyVTze6SmGs1DHtKa

If you still have any issues, would it be possible to send us your (invalid) IPA file along with one of the libswift*.dylib files that you have been linking against? via https://transfer.harman.com/requests/OzchDTANAktlPIdnom98pO

thanks

**Comment by chrslgr** [comment]
Hi Guys, hi @ajwfrost.

I too came across this problem when trying to build an ipa for AppStore (MacOS 10.14, Intellij 2020.2.4, Air 33.1.1217).

But I don't even come as far as getting an .ipa because Intellij stops the build process due to the error:
`Failed to package AIR application MyApp.ipa: Warning: Using default implementation for native extension 'androidx.core' The signing certficate is not from the expected issuer`

We use several external ANEs (distriqt/myflashlabs) - do they require resigning? 
Do you have a hint for me how to fix this?

Thanks a lot
Chris

**Comment by ajwfrost** [comment]
Hi @chrslgr 

The two lines there probably mean:

1. default implementation for androidx.core -> if ADT can't find the right platform, which would be e.g. iPhone-ARM, then this falls back on the 'default' (AS3-only) implementation. I don't know for sure but I would wonder whether that ANE should be included in an IPA build..?
2. signing certificate not being from the expected issuer is where we started with this one ... are you just using AIR 33.1.1.217? in which case yes, the certificate that you're using may be a later one and in the older AIR builds, we didn't have Apple's newer intermediate certificates that are the issuers of your new one. So you would need to grab the latest adt.jar file from the above comment (although beware, if you're building for Android then it will only work if the rest of the SDK is 33.1.1.406)

thanks
  Andrew

**Comment by marchbold** [comment]
Just a comment on the `Warning: Using default implementation for native extension 'androidx.core'` part. 

This simply means that the extension doesn't have an implementation for the platform you are packaging for and will use the default. I believe it is misleading, however it is "warning" you incase you thought it did have an implementation. 

There is no reason for an Android only ANE such as `androidx.core` to include an iOS build. 

This part of the message is completely unrelated to the signing issue.

**Comment by chrslgr** [comment]
@ajwfrost @marchbold 
thanks to you both. 
switching from air 33.1.1.217 to 33.1.1.406 solved the error with the ANE signature, copying the adt.jar into the sdk removed a further problem with Apple Transporter ('...is not signed in using an Apple submission certificate').
ipa was built, and app installed successfully via TestFlight.
Thanks!

**Comment by toysonlocation** [comment]
I've been having the same problem as Dima107 above

WARNING: could not locate iTunesMetadata.plist in archive!
WARNING: could not locate Payload/Test.app/SC_Info/Test.sinf in archive!
ERROR: Install failed. Got error "ApplicationVerificationFailed" with code 0xe8008018: Failed to verify code signature of /var/installd/Library/Caches/com.apple.mobile.installd.staging/temp.gm4pOV/extracted/Payload/Test.app : 0xe8008018 (The identity used to sign the executable is no longer valid.)

I'm using 33.1.1.406 in Catalina, and have put the most recent adt.jar into service but with no change of error. Please advise!

**Comment by ajwfrost** [comment]
Hi @toysonlocation  - I'm assuming that the error message shown isn't actually the case (ADT should have picked up on this anyway prior to that..) so could you please upload your generated IPA file to the below link and we can inspect the signatures to see what the issue might be?
https://transfer.harman.com/requests/5XlmiBiv9xy5vgSrVD1h4D

many thanks

**Comment by pavelzmolik** [comment]
> Hi @megajogos @pavelzmolik
> Thanks for the feedback: we have another update to ADT.jar that should fix the signing of the Swift libraries:
> https://transfer.harman.com/message/Am40qkyVTze6SmGs1DHtKa
> 
> If you still have any issues, would it be possible to send us your (invalid) IPA file along with one of the libswift*.dylib files that you have been linking against? via https://transfer.harman.com/requests/OzchDTANAktlPIdnom98pO
> 
> thanks

Hello @ajwfrost With the new ADT.jar my bussines partner confirmed me that Apple store doesnt reject new IPA compiled with current SDK v406 and the newest ADT you provide and distribution certificates.. so I am glad to thank you for your work..!

**Comment by ajwfrost** [comment]
@pavelzmolik  that's good news, thanks for confirming!

@toysonlocation could you send me a direct email as I want to check something re. your certificate. There's a field missing when I compare this with our ones (extension id 1.2.840.113635.100.6.1.12) and it's actually got a different issuer, even though the name is the same (which is why it passes our initial test and doesn't say that it's not from the expected issuer...)

thanks

**Comment by toysonlocation** [comment]
Hi Andrew,

Let me know what you need

Best,
Joseph

**Comment by Jasenvm** [comment]
Hi All
I'm having a similar issue as above, but have not had time to research to fix, considering this issue is still active....but I did find and act on this by our good friends at Apple (installed this https://www.apple.com/certificateauthority/AppleWWDRCAG3.cer)

Maybe this helps with the issuer issue moving forward?

https://developer.apple.com/support/expiration/

**Comment by YasukazuU** [comment]
Hello
I'm pleased to inform you that ipa using the solution here has passed Apple's review
I built it with 33.1.1.406 and the latest adt.jar
I used the newly registered certificate this month

Great, thank you

**Comment by ac27437676** [comment]
> Hello
> I'm pleased to inform you that ipa using the solution here has passed Apple's review
> I built it with 33.1.1.406 and the latest adt.jar
> I used the newly registered certificate this month
> 
> Great, thank you

I have same problem，how can you deal it？

**Comment by YasukazuU** [comment]
I was getting the same error
I received an error on Big Sur and noticed that itunes is obsolete
Successful build on Mojave with itunes installed
I don't know if this will be helpful

**Comment by toysonlocation** [comment]
> Hi @toysonlocation - I'm assuming that the error message shown isn't actually the case (ADT should have picked up on this anyway prior to that..) so could you please upload your generated IPA file to the below link and we can inspect the signatures to see what the issue might be?
> https://transfer.harman.com/requests/5XlmiBiv9xy5vgSrVD1h4D
> 
> many thanks

Thank @ajwfrost for your direct help. After again, but with much greater certitude, redoing all certs and keys, I have a test app deploying to the device without error. I don't know what went wrong, but it was probably just a PICNIC all along. Thanks again!

**Comment by Umanoff** [comment]
Just checking, before I try to build with 33.1.1.406. We think that this is working on both Windows _and_ Mac, correct?

**Comment by ajwfrost** [comment]
@Umanoff yes the additional ADT.jar file from above is platform-agnostic, so it will produce the same binaries whether you run this from a Windows or Mac development machine.

@toysonlocation thanks for your response and efforts there. Sounds like our latest update is covering all scenarios now so I'll kick off the release process, will just see if we have any other fixes that we can squeeze in..

thanks

**Comment by mc3dev** [comment]
Thank you, @ajwfrost, for the fix for the adt.ja for AIR SDK 33.1.1.406. I am able to publish the IPA with new certificate now. However, I am running an new problem related to an NativeWebView ANE done by Disqrit. It was not a problem from the earlier version of AIR for iOS SDK but started now after I published the IPA with the certificate fix. It made me think this is something in the AIR SDK with iOS webkit, not on the ANE side. Therefore I am posting the issue here also.

Before the AIR SDK fix, the webview window set to fit the stage size loads and display the HTML well. HTML is 1024x768. Stage size is 1024x768, the webview window is also 1024x768.  But now with the new AIR SDK, the webview window does not line up with the stage size, and the HTML content inside the webview does not scale to fit the webview.
This is what it looks like on iPad Pro 12.9" 4th gen - 
![image](https://user-images.githubusercontent.com/5835724/110360977-660d2600-800d-11eb-9d48-c76c21dd122e.png)

Here is what it looks like on iPad 10.2" 7th Gen -
![image](https://user-images.githubusercontent.com/5835724/110360999-6dccca80-800d-11eb-9bf1-0d695537c2ef.png)

Expected behaviour
HTML page (with graphic 1024x768) loaded via the webview fills the webview window because the webview window is set by the stage width and stage height (the screenshot is taken from the app published via AIR SDK 33.0.2.246)

Can you please look into this and provide feedback/advice?

**Comment by ajwfrost** [comment]
@mc3dev looks curious (and maybe related to start-up/launchscreen changes in the SDK) - but could you please raise this in a new issue, as this current one will be closed once we've published the updated SDK with the fix for the signing problems.

thanks

**Comment by mc3dev** [comment]
> @mc3dev looks curious (and maybe related to start-up/launchscreen changes in the SDK) - but could you please raise this in a new issue, as this current one will be closed once we've published the updated SDK with the fix for the signing problems.
> 
> thanks

Just added a new issue  #719. Thank you!

**Comment by BotikiGames** [comment]
> > Hello
> > I'm pleased to inform you that ipa using the solution here has passed Apple's review
> > I built it with 33.1.1.406 and the latest adt.jar
> > I used the newly registered certificate this month
> > Great, thank you
> 
> I have same problem，how can you deal it？

I have same problem! 
ERROR ITMS-90179: "Invalid Code Signing. The executable 'Payload/****.app/****' must be signed with the certificate that is contained in the provisioning profile."
new adt.jar + SDK 33.1.1.406 + new distribution certificate
Do you have any solution for this issue?

**Comment by marchbold** [comment]
@BotikiGames Have you double checked that you downloaded a new profile after updating your certificate?

**Comment by BotikiGames** [comment]
> Have you double checked that you downloaded a new profile after updating your certificate?

hm, after removing the old Apple Worldwide Developer Relations Certificate (valid until 2023) and installing a new certificate (valid until 2030) my IPA was accepted in Appstore.
Everything works! Thanks to everyone working on AIR.

**Comment by MatseFR** [comment]
Hello, I'm trying to update an existing app using the latest ADT.jar provided here : it works fine if I do a test build but when trying to package an adhoc version I get the following error during install

WARNING: could not locate iTunesMetadata.plist in archive!
WARNING: could not locate Payload/PANews.app/SC_Info/PANews.sinf in archive!
ERROR: Install failed. Got error "ApplicationVerificationFailed" with code 0xe80
08016: Failed to verify code signature of /private/var/installd/Library/Caches/c
om.apple.mobile.installd.staging/temp.WB6fyU/extracted/Payload/PANews.app : 0xe8
008016 (The executable was signed with invalid entitlements.)

I tried installing through testfairy and it fails as well...

I just re-created every certificate and provisionning profile (and did that already a few days ago) on windows using openssl, pretty sure they're all good. I can send the IPA if needed

**Comment by ajwfrost** [comment]
Hi - thanks for letting us know, we'll check on the entitlements.. Yes please if you could upload your IPA it would be handy. If you have a Mac and are able to re-sign it, using the workaround mentioned earlier, and send a working version as well it would be useful. Plus do you have any additional entitlements being added by the app descriptor file?
https://transfer.harman.com/requests/ocIfBYZSr8bJpt4jMzjlHQ

thanks

**Comment by MatseFR** [comment]
Thanks for the quick reply ! Transfer sent with the details.

For the record, I have entitlements for push notifications :

<Entitlements><![CDATA[
	
	
	<key>get-task-allow</key>
	<true/>
	<key>aps-environment</key>
	<string>development</string>
	
	
	
	
	
	<key>application-identifier</key>
	<string>xxxxxxxxxx.air.PANews</string>
	<key>keychain-access-groups</key>
	<array>
		<string>xxxxxxxxxx.*</string>
	</array>
	
]]></Entitlements>

**Comment by MatseFR** [comment]
Oh wait ! Seems the problem was on my side : I thought I had to keep the dev aps environment for adhoc but just tried with production instead and it works !

Problem solved, my apologies for the inconvenience !

**Comment by ajwfrost** [comment]
@MatseFR okay great! thanks for the update :-)

**Comment by zishani** [comment]
Hi there, where can I get hold of the patched adt.jar to use with 406? The above links have all expired...

Thanks

**Comment by marchbold** [comment]
He posted this link in #521 
https://transfer.harman.com/message/EOmfgAn5M3SPSACcUPuG4I

**Comment by spences** [comment]
I am having the same issue using FB4.5 with SDK 4.5.1 and Air 27 for my ios app. I am using a windows 10 environment. All was well until I tried to create a new .ipa file with Apple new certificate. I need help in resolving this issue as soon as possible. Does Air 33.xxxxx solve this issue?

**Comment by ajwfrost** [comment]
Well with AIR 27 you would definitely get this problem. The error message "the signing certificate is not from the expected issuer" happens when the common name of the "issuer" that's listed in your signing certificate does not match with the Apple intermediate certificate that is included in the AIR packaging tool. When Apple started using this different intermediate certificate as the issuer for people's certificates, it broke this check in AIR so with build 33.1.1.406 we included the new intermediate certificate. So this error shouldn't be seen any more: if it is, it implies there's a different intermediate certificate that's been used, and we'd need to know what that is to see if we can add that to our list of 'approved' Apple intermediates.

There was then the secondary problem with code-signing formats where Apple no longer accepted the older code sign mechanism so we had to update this in ADT, which took some time to get right but I believe we are almost there - most things work, there are issues with Swift libraries still that we're working to resolve shortly.

So - short answer: yes I would hope the latest AIR SDK builds will solve the certificate problems for IPA files, but there are a couple of open items still..

**Comment by spences** [comment]
Thank you, Andrew. Greatly appreciated.

Stewart

Stewart Spence
Acting Manager of Information Technology Services, Focussing on the SIS transition.
Thames Valley District School Board
519-452-2000 ext. 20138
***@***.******@***.***>

Confidentiality Warning:  This message and any attachments are intended only for the use of the intended recipient(s) and may contain confidential or personal information that may be subject to the provisions of the Municipal Freedom of Information and Protection of Privacy Act.  If you are not the intended recipient or an authorized representative of the intended recipient, you are notified that any dissemination of this communication is strictly prohibited.  If you have received this communication in error, please notify the sender immediately and delete the message and any attachments.

**Comment by ajwfrost** [comment]
Closing this issue as we believe we have now got to a point with the latest AIR SDK where all the code-signing is working for iOS on both mac and windows platforms... for these sorts of problems, it may be best to raise a new, specific issue if there is still a problem with it.
thanks
